sdk/python/feast/data_source.py

```diff
@@ -767,6 +769,21 @@ def get_table_column_names_and_types(self) -> Iterable[Tuple[str, str]]:
 
         return name_type_pairs
 
+    def _assert_table_exists(self):
+        from google.api_core.exceptions import NotFound
+        from google.cloud import bigquery
+
+        client = bigquery.Client()
+
+        try:
+            client.get_table(self.table_ref)
+        except NotFound:
+            raise NameError(
+                f"""
+                Unable to find table {self.table_ref} in BigQuery. Please check that table exists.
```

Comment:
Just swooping by here. Don't have all the context. But if the idea is that we fail based on an upstream table being missing, then would it make more sense to
* Use a custom error type
* Make sure that our tests test all data sources (BigQuery and File sources) and ensure that they raise this custom exception on a missing source?

Reply:
Good idea! Let me know if the changes in https://github.com/feast-dev/feast/pull/1627/commits/f15f3f61865e7dadbbdca41a9e6ad755008ef93f are what you meant.

Reply:
I tried to check FileSources in https://github.com/feast-dev/feast/pull/1627/commits/f15f3f61865e7dadbbdca41a9e6ad755008ef93f but it ended up being more complicated than I had expected - seems like a lot of tests were failing maybe because they use temporary files. Either way, it seems like a larger change that could be externalized in a separate issue/PR.

I did adopt the change for a custom exception though, thanks for the feedback!